thirdparty-api/data-models.md

```diff
@@ -0,0 +1,953 @@
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| serviceType | 1 | Enum of String(1..32) | See [Section 3.2.2.10](#3229-servicetype) below ServiceType) for more information on allowed values. |
+
+##### 3.2.1.25 TransactionType
+The TransactionType type used in these definitions is as defined in [Section 7.4.18](https://github.com/mojaloop/mojaloop-specification/blob/master/fspiop-api/documents/v1.1-document-set/API%20Definition%20v1.1.md#7418-transactiontype) of Ref. 1 above.
+##### 3.2.1.26 TransactionState
+The TransactionState type used in these definitions is as defined in [Section 7.3.33](https://github.com/mojaloop/mojaloop-specification/blob/master/fspiop-api/documents/v1.1-document-set/API%20Definition%20v1.1.md#7333-transactionstate) of Ref. 1 above.
+##### 3.2.1.27 Uri
+The API data type Uri is a JSON string in a canonical format that is restricted by a regular expression for interoperability reasons. The regular expression for restricting the Uri type is as follows:
+`^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))? `
+
+##### 3.2.1.28 ErrorInformation
+
+For details, see section [7.4.2 ErrorInformation](../fspiop-api/documents/API-Definition_v1.1.md#742-errorinformation) in the Mojaloop FSPIOP API Definition.
+
+##### 3.2.1.29 FIDOPublicKeyCredentialAttestation
+
+A data model representing a FIDO Attestation result. Derived from [`PublicKeyCredential` Interface](https://w3c.github.io/webauthn/#iface-pkcredential),
+where the ArrayBuffer fields are represented as base64-encoded utf-8 strings.
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| id       | 1    | string | The identifier of a keypair created by an authenticator |
+| rawId    | 0..1 | string | The identifier of a keypair created by an authenticator, base64 encoded |
+| response | 1    | AuthenticatorAttestationResponse | The attestation response from the authenticator |
+
+##### 3.2.1.30 FIDOPublicKeyCredentialAssertion
+
+A data model representing a FIDO Assertion result. Derived from [`PublicKeyCredential` Interface](https://w3c.github.io/webauthn/#iface-pkcredential),
+where the ArrayBuffer fields are represented as base64-encoded utf-8 strings.
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| id       | 1    | string | The identifier of a keypair created by an authenticator |
+| rawId    | 0..1 | string | The identifier of a keypair created by an authenticator, base64 encoded |
+| response | 1    | AuthenticatorAssertionResponse | The assertion response from the authenticator |
+
+##### 3.2.1.31 AuthenticatorAttestationResponse
+
+A data model representing an [AttestationStatement](https://w3c.github.io/webauthn/#attestation-statement).
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| clientDataJSON    | 1 | string | JSON string with client data |
```

Comment:
> | clientDataJSON    | 1 | string | JSON string with client data |

Please use a max length of the string (https://apisecurity.io/encyclopedia/content/oasv2/datavalidation/parameters/parameter-string-maxlength)